docs/docs/migrate-data.md

```diff
@@ -0,0 +1,213 @@
+
+If your plugin saved the User ID (the ID encoded in the authentication token) in the `profiles` array on the `user` document along with a `provider` field, for example:
+
+```json
+{
+  "profiles": [
+    {
+      "provider": "my-auth",
+      "id": "..."
+    }
+  ]
+}
+```
+
+then you will need to configure the `CORAL_MAPPER_USERS_SSO_PROVIDER` environment variable with the name of the `provider`.
+
+```bash
+export CORAL_MAPPER_USERS_SSO_PROVIDER="my-auth"
+```
+
+---
+
+Insert note about how to configure mapper if the ID field is not contained in the profiles array
+
+---
+
+If your custom plugin wrote the username field to a field that is _not_ called `username` at the top-level of the `user` document, then you will need to set the `CORAL_MAPPER_USERS_USERNAME` environment variable. For example, if your data looks like this:
+
+```json
+{
+  "metadata": {
+    "displayName": "user name"
+  }
+}
+```
+
+you would set `CORAL_MAPPER_USERS_USERNAME="metadata.displayName`. If your username value is stored in `username` at the root level of the document, you do not need to configure this value.
+
+The following example script will run the mapper step as a "dry run" to validate the data:
+
+```bash
+
+# Set this to the folder that contains the exported documents from v4
+export CORAL_INPUT_DIRECTORY="$PWD/coral/input"
+
+# Set this to the folder containing the transformed documents
+export CORAL_OUTPUT_DIRECTORY="$PWD/coral/output"
+
+# Set this to the ID of your new v7 Tenant.
+export CORAL_TENANT_ID=""
+
+# Set this to the ID of your v7 Site.
+export CORAL_SITE_ID=""
+
+export CORAL_LOG="$PWD/coral/logs.json"
+
+# Set this to the name of a directory we can write output files that have
+# been mapped.
+export CORAL_MAPPER_POST_DIRECTORY="$PWD/coral/post"
+
+mkdir -p "${CORAL_MAPPER_POST_DIRECTORY}"
+
+# see note above, uncomment if necessary
+# export CORAL_MAPPER_USERS_SSO_PROVIDER=""
+
+# see note above, uncomment if necessary
+# export CORAL_MAPPER_USERS_USERNAME=""
+
+coral-importer legacy --dryRun map
+
+```
+
+If this script succeeded, run it a second time, removing the `--dryRun` flag. If it did not succeed, inspect the logs.json file for more information.
+
+If you have multiple values for `provider` in your dataset, you will need to complete this step multiple times, once for each provider. For example, if you have two `provider` values, `auth-1` and `auth-2`, you will first set `CORAL_MAPPER_USERS_SSO_PROVIDER` to `auth-1`, run the `map` command. Once this has succeeded, replace the `users.json` file in the output directory with the `users.json` file created by the `map` command. Then set `CORAL_MAPPER_USERS_SSO_PROVIDER` to `auth-2` and run the command again.
+
+Once you have completed the import mapping process, replace the `users.json` file in your output directory with the `users.json` file created by the mapper and move on to the next step.
```

Comment:
Since this outputs a new `users.json` every time, how can it be updating the provider on the users? It has no recollection of the previous run, so it doesn't know to "append a provider" to an existing user.

I would run this by Wyatt, if you need clarification, feel free to pull me into the discussion as I believe this is an edge-case we forgot to address.

Reply:
Discussed this with Wyatt. You can't validate multiple v4 auth providers into multiple v6 SSO profiles. 

An example to better demonstrate the way you'd port from v4 to v6:

- We have 100 users
- 30 users are set up with an SSO provider `A`.
- 70 remaining users are set up with an SSO provider `B`.
- Ensure that NO users have both provider `A` AND provider `B` in their v4 profiles or you'll get duplicate users in the next step

We need to do the following to make this work:

- Run `coral-import-mapper` targeting provider A
   - set the output dir to `providerA/users.json`
- Run the `coral-import-mapper` target provider B
   - set the output dir to `providerB/users.json`
- Combine the two users.json together

So, what happens if I have users with 2 or more v4 SSO profiles? Don't I want to map them to 2 v6 SSO profiles? 
**A:** No, you don't really want to do this. Pick one SSO provider for v6 and stick with it.

What if I have two providers (`A` and `B`) and the majority are using `A`, but some are using `B`?
**A:** You'll have to manually handle these users by doing the following:

- Run `coral-import-mapper` targeting provider B
   - set the output dir to `providerB/users.json`
- Manually update the resulting json file to change the v6 profiles to provider A
   - help these users migrate over to A by telling them to login using auth provider A in v6 to re-map their account ot the newly made provider A user account